src/cpp/rtps/reader/BaseReader.cpp

```diff
@@ -272,6 +279,11 @@ void BaseReader::allow_unknown_writers()
     accept_messages_from_unkown_writers_ = true;
 }
 
+LocalReaderPointer BaseReader::get_local_pointer()
+{
+    return LocalReaderPointer(this, local_view_);
```

Comment:
A comment here: I haven't taken the reader's mutex before the `return` since tsan reported one potential deadlock in the `LatencyTest` between that one and the rtps endpoint collection mutex. I think it is fair (and quick) enough copying the shared pointer without protection.